types/test/findOne.ts

```diff
@@ -1,6 +1,17 @@
+import { expectTypeOf } from 'expect-type'
 import { User } from "./models/User";
 
-User.findOne({ where: { firstName: 'John' } });
+async () => {
+    const user = await User.findOne({ where: { firstName: 'John' } });
+    expectTypeOf(user).toEqualTypeOf<User | null>()
 
-// @ts-expect-error
-User.findOne({ where: { blah: 'blah2' } });
+    // @ts-expect-error
```

Comment:
```suggestion
    // @ts-expect-error blah is not an attribute of User
```

Reply:
Added to base PR https://github.com/sequelize/sequelize/pull/12921/commits/9abadb1781610db10e16f308aac6131e4672a6c5